modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java

```diff
@@ -46,26 +46,41 @@
 
 	private static Log log = LogFactory.getLog(OperationMgtService.class);
 
-	@GET
+	@PUT
 	@Produces(MediaType.APPLICATION_JSON)
 	@Path("{id}")
 	public List<? extends org.wso2.carbon.device.mgt.common.operation.mgt.Operation> getPendingOperations
-			(@HeaderParam("Accept") String acceptHeader, @PathParam("id") String id) {
+			(@HeaderParam("Accept") String acceptHeader, @PathParam("id") String id,
+			 List<? extends org.wso2.carbon.device.mgt.common.operation.mgt.Operation> resultOperations) {
 
+		if (log.isDebugEnabled()) {
+			log.debug("Invoking Android pending operations:" + id);
+		}
 		Message message = new Message();
 		MediaType responseMediaType = AndroidAPIUtils.getResponseMediaType(acceptHeader);
-		java.util.List<? extends org.wso2.carbon.device.mgt.common.operation.mgt.Operation> operations = new
-				ArrayList<Operation>();
-		if (log.isDebugEnabled()) {
-			log.debug("Invoking Android pending operations:"+id);
+
+		try {
+			if (resultOperations != null) {
+				updateOperations(resultOperations);
+			}
+		} catch (OperationManagementException e) {
+			message.setResponseMessage("Issue in retrieving operation management service instance");
+			log.error(message.getResponseMessage(), e);
 		}
+
 		DeviceIdentifier deviceIdentifier = AndroidAPIUtils.convertToDeviceIdentifierObject(id);
+		List<? extends org.wso2.carbon.device.mgt.common.operation.mgt.Operation> operations;
+
 		try {
-			operations = AndroidAPIUtils.getDeviceManagementService().getPendingOperations(deviceIdentifier);
-		}catch(OperationManagementException ex){
-			message.setResponseMessage("Issue in retrieving operation management service instance");
+			operations = AndroidAPIUtils.getPendingOperations(deviceIdentifier);
+		} catch (OperationManagementException e) {
+			String errorMessage = "Issue in retrieving operation management service instance";
+			message.setResponseMessage(errorMessage);
```

Comment:
Looks like we're setting too many attributes upon the instance of "Message". Perhaps, it's good to use "BuilderPattern" or something similar to make the code less bulky.